### PR TITLE
[misc] Enable offline cache by default (TEST)

### PR DESCRIPTION
**Pull request by PGZXB**
Related issue = #4401

### CODE REVIEW

**Comment by lin-hitonami** [comment]
We may need to repeat the tests for CPU and CUDA in these files to ensure that the offline cache works fine.
https://github.com/taichi-dev/taichi/blob/master/.github/workflows/scripts/win_test.ps1
https://github.com/taichi-dev/taichi/blob/master/.github/workflows/scripts/unix_test.sh

File: taichi/runtime/program_impls/llvm/llvm_program.cpp [hunk 1]
```diff
@@ -100,8 +100,16 @@ std::unique_ptr<AotModuleBuilder> LlvmProgramImpl::make_aot_module_builder() {
...
+  // `offline_cache=true`. In the future, we will refactor the implementations
+  // of LLVM AOT and Offline Cache to fix the problem better.
+  auto saved_offline_cache_config = kernel.program->config.offline_cache;
+  kernel.program->config.offline_cache = false;
   auto compiled_fn =
       this->compile(&kernel, nullptr);  // Offloaded used in async mode only
+  kernel.program->config.offline_cache = saved_offline_cache_config;
 
   const std::string &kernel_key = kernel.get_cached_kernel_key();
   TI_ASSERT(cache_data_->kernels.count(kernel_key));
```
Comment (lin-hitonami) [inline taichi/runtime/program_impls/llvm/llvm_program.cpp:115]: Can we get the kernel data directly from the offline cache?

**Comment by lin-hitonami** [comment]
BTW I suggest we fix the bug in another PR.

File: taichi/runtime/program_impls/llvm/llvm_program.cpp [hunk 2]
```diff
@@ -100,8 +100,16 @@ std::unique_ptr<AotModuleBuilder> LlvmProgramImpl::make_aot_module_builder() {
 }
 
 std::unique_ptr<aot::Kernel> LlvmProgramImpl::make_aot_kernel(Kernel &kernel) {
+  // TODO(PGZXB): Temporarily set `offline_cache=false` to make
+  // `KernelCodeGen::codegen` generate compiled data correctly, which is a
+  // temporary solution to avoid make_aot_kernel failed when
+  // `offline_cache=true`. In the future, we will refactor the implementations
+  // of LLVM AOT and Offline Cache to fix the problem better.
+  auto saved_offline_cache_config = kernel.program->config.offline_cache;
```
Comment (lin-hitonami) [inline taichi/runtime/program_impls/llvm/llvm_program.cpp:108]: Is this still necessary?

**Comment by taichi-gardener** [comment]
Rebase failed, please resolve the conflicts manually.

**Comment by taichi-gardener** [comment]
Rebase failed, please resolve the conflicts manually.

**Comment by taichi-gardener** [comment]
Rebase failed, please resolve the conflicts manually.

**Pull request closed without merging**